healthcheck/healthcheck_test.go

```diff
@@ -193,6 +193,64 @@ func TestSetBackendsConfiguration(t *testing.T) {
 	}
 }
 
+func TestNewRequest(t *testing.T) {
+	tests := []struct {
+		host     string
+		port     int
+		path     string
+		expected string
+	}{
+		{
+			host:     "backend1:80",
+			port:     0,
+			path:     "/test",
+			expected: "http://backend1:80/test",
+		},
+		{
+			host:     "backend2:80",
+			port:     8080,
+			path:     "/test",
+			expected: "http://backend2:8080/test",
+		},
+		{
+			host:     "backend1",
+			port:     0,
+			path:     "/health",
+			expected: "http://backend1/health",
+		},
+		{
+			host:     "backend2",
+			port:     8080,
+			path:     "/health",
+			expected: "http://backend2:8080/health",
+		},
+	}
+
+	for _, test := range tests {
+		backend := NewBackendHealthCheck(
```

Comment:
Please use parallelized sub-tests here. See [one of the existing tests](https://github.com/containous/traefik/blob/acc7865542e0558de4de47b18c3acbe5d51d4b5c/healthcheck/healthcheck_test.go#L140-L142) for what needs to be changed for this (in a nutshell:

- `test := test`
- `t.Run(...)`
- `t.Parallel()`

)